src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1337,6 +1337,12 @@ EnsureTableNotDistributed(Oid relationId)
 static void
 EnsureRelationHasNoTriggers(Oid relationId)
 {
+	if (EnableUnsafeTriggers)
+	{
+		/* user really wants triggers */
+		return;
+	}
+
 	List *explicitTriggerIds = GetExplicitTriggerIdList(relationId);
 
 	if (list_length(explicitTriggerIds) > 0)
```

Comment:
maybe,

```c
		ereport(ERROR, (errmsg("cannot distribute relation \"%s\" because it has "
							   "triggers and \"citus.enable_unsafe_triggers\" is "
                               "set to \"false\"", relationName),
						errhint("Consider setting \"citus.enable_unsafe_triggers\" "
                                "to \"true\", or drop all the triggers on \"%s\" "
                                "and retry.", relationName)));
```